ctapipe/image/time_next_neighbor_cleaning.py

```diff
@@ -180,14 +205,15 @@ def calculate_ipr_from_source(self, source, calibrator, charge_steps):
 
         """
         for event in source:
```

Comment:
You should probably skip all events that are not of type `SKY_PEDESTAL` like so:

```
if event.trigger.type != EventType.SKY_PEDESTAL:
    continue
```